akka-remote/src/main/scala/akka/remote/AddressUidExtension.scala

```diff
@@ -35,13 +33,7 @@ class AddressUidExtension(val system: ExtendedActorSystem) extends Extension {
 
   private def arteryEnabled = system.provider.asInstanceOf[RemoteActorRefProvider].remoteSettings.Artery.Enabled
 
-  val longAddressUid: Long = {
-    val tlr = ThreadLocalRandom.current
-    if (arteryEnabled) tlr.nextLong()
-    // with the old remoting we need to make toInt.toLong return the same number
-    // to keep wire compatibility
```

Comment:
we need to keep this compatibility, so some config check at the ActorSystem startup is needed